spec/compiler/macro/macro_methods_spec.cr

```diff
@@ -1866,6 +1866,126 @@ module Crystal
         end
       end
 
+      describe "#constant" do
+        it "global path" do
+          assert_type(%(
+            ID = 10
+
+            class A
+              class B
+              end
+            end
+
+            {{ A::B.constant("::ID") == 10 ? 1 : 'f' }}
+          )) { int32 }
+        end
+
+        it "global path with extra ::" do
+          assert_macro_error %({{@type.constant "::::ID" }}), %(Invalid constant name: "::::ID")
```

Comment:
It's enough weird characters together :P

```suggestion
          assert_macro_error %({{ @type.constant "::::ID" }}), %(Invalid constant name: "::::ID")
```

Reply:
Really, I think this looks ill-formatted with whitespace on the right, but none on the left.
Ditto below.